### DISCUSSION TITLE
Inertia with Laravel Policy - Getting a modal with 403 error

### DISCUSSION THREAD

**Discussion by pmochine** [inertiajs/inertia#835] [Help]
Hey stupid question, but I'm not sure how to handle this correctly.

I have installed  Laravel Jetstream with Inertia and added a middleware for the deletion of a user. 

    Route::delete('/user', [CurrentUserController::class, 'destroy'])
                ->name('current-user.destroy')
                ->middleware('can:delete,App\Models\User');

Now in my UserPolicy I have added:

    public function delete(User $user)
    {
        if ($user->hasActiveSubscription()) {
            // User is still active payer
            return Response::deny(__('policy.active_subscription'));
        }
        ...
        return Response::allow();
    }

How can I combine Inertia with Laravel Policy?

Currently, I'm just getting a 403 modal popup.

Thank you 🤗 

Edit: 
Ok I could just use "can" like here https://inertiajs.com/authorization, but I hoped for a more lazy approach. Similar like in validation 😅

**Comment by mlmendes** [comment]
Go to the boostrap/app.php file and on the withExceptions() function, add a handler for HTTP exceptions, like this:

```
->withExceptions(function (Exceptions $exceptions): void {
        $exceptions->render(function (HttpExceptionInterface $e, Request $request) {
            if (!$request->expectsJson() && $e->getStatusCode() === 403) {
                return redirect()->back()->with('flash.error', __('This action is unauthorized.'));
            }
            return $e->getMessage();
        });
    })
```

Then do whatever you want with this flash data. I added it to the shared Inertia data on app/Http/Middleware/HandleInertiaRequests.php

```
public function share(Request $request): array
    {
        [$message, $author] = str(Inspiring::quotes()->random())->explode('-');

        return [
            ...parent::share($request),
            'name' => config('app.name'),
            'quote' => ['message' => trim($message), 'author' => trim($author)],
            'auth' => [
                'user' => $request->user(),
            ],
            'sidebarOpen' => !$request->hasCookie('sidebar_state') || $request->cookie('sidebar_state') === 'true',
            'flash' => [
                'error' => fn() => $request->session()->get(key: 'flash.error'),
                'info' => fn() => $request->session()->get(key: 'flash.info'),
                'success' => fn() => $request->session()->get(key: 'flash.success'),
                'warning' => fn() => $request->session()->get(key: 'flash.warning'),
            ],
        ];
    }
```

This shared data is available on the Inertia font-end side.
